WordPress/src/main/java/org/wordpress/android/ui/suggestion/service/SuggestionService.java

```diff
@@ -97,11 +100,68 @@ public void run() {
         }.start();
     }
 
-    private void removeSiteIdFromRequestsAndStopServiceIfNecessary(Integer remoteBlogId) {
-        mCurrentlyRequestingSiteIds.remove(remoteBlogId);
+    private void removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(Integer remoteBlogId) {
+        mCurrentlyRequestingSuggestionsSiteIds.remove(remoteBlogId);
 
         // if there are no requests being made, we want to stop the service
-        if (mCurrentlyRequestingSiteIds.isEmpty()) {
+        if (mCurrentlyRequestingSuggestionsSiteIds.isEmpty()) {
+            AppLog.d(AppLog.T.SUGGESTION, "stopping suggestion service");
+            stopSelf();
+        }
+    }
+
+    public void updateTags(final int remoteBlogId) {
+        if (mCurrentlyRequestingTagsSiteIds.contains(remoteBlogId)) {
+            return;
+        }
+        mCurrentlyRequestingTagsSiteIds.add(remoteBlogId);
+        RestRequest.Listener listener = new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject jsonObject) {
+                handleTagsUpdatedResponse(remoteBlogId, jsonObject);
+                removeSiteIdFromTagRequestsAndStopServiceIfNecessary(remoteBlogId);
+            }
+        };
+        RestRequest.ErrorListener errorListener = new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError volleyError) {
+                AppLog.e(AppLog.T.SUGGESTION, volleyError);
+                removeSiteIdFromTagRequestsAndStopServiceIfNecessary(remoteBlogId);
+            }
+        };
+
+        AppLog.d(AppLog.T.SUGGESTION, "suggestion service > updating tags for siteId: " + remoteBlogId);
+        String path = "/sites/" + remoteBlogId + "/tags";
+        WordPress.getRestClientUtils().get(path, listener, errorListener);
```

Comment:
That won't work on self hosted sites (they only support XMLRPC API calls). We should fill a new issue for this.